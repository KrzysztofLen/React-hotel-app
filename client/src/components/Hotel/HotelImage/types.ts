export type Props = {
  onClick: () => void;
  image: any;
};
