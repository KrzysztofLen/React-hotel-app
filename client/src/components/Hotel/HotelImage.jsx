import React from 'react';

const HotelImage = (data) => <img onClick={data.onClick} className="hotel__image" src={data.image} alt="xyz"/>;

export default HotelImage;