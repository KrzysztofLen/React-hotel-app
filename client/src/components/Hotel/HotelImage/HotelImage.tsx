import * as React from 'react';
import noImage from '../../../assets/img/No_Image_Available.jpg';

interface IProps {
    onClick: () => void,
    image: any
}

const HotelImage = (data: IProps) => {
    console.log(data);
    return (<React.Fragment>
            <div className="overlay" onClick={data.onClick}>
                <span className="overlay__text">Click</span>
            </div>
            {(data.image == null) ? <img src={noImage} alt="zyx"/> :
                <img className="hotel__image" src={data.image[0]} alt="xyz"/>}
        </React.Fragment>
    )
};

export default HotelImage;