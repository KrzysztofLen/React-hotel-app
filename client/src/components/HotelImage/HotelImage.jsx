import React from 'react';

const HotelImage = (data) => <img onClick={data.onClick} className="hotel__image" src={data.image[0]} alt="xyz"/>;

export default HotelImage;