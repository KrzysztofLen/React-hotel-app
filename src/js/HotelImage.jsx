import React, {Component} from 'react';

const HotelImage = (data) => <img onClick={data.onClick} className="hotel__image" src={data.image} alt="xyz"/>;

export default HotelImage;