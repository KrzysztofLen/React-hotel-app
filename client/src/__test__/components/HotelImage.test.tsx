import React from 'react';
import { shallow } from 'enzyme';
import HotelImage from '../../components/Hotel/HotelImage/HotelImage';

const app_data = [
  {
    id: 0,
    name: 'Test Eva',
    hotel_name: 'Test Sunset Willow Resort',
    image: 'https://preview.ibb.co/ivpH07/alex_block_354270_unsplash.jpg',
    rate: 5,
    more_activities: true,
    activities_number: 6,
    commented: 5,
    peers: 46,
    discussions: 29,
    findings: 19,
    questions: 10,
    activity_level: 2,
  },
];

describe('HotelImage component', () => {
  it('should render HotelImage component correctly', () => {
    const wrapper = shallow(
      <HotelImage image={app_data[0].image} onClick={() => {}} />,
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <HotelImage onClick={onClick} image={app_data[0].image} />,
    );
    wrapper.find('div').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
