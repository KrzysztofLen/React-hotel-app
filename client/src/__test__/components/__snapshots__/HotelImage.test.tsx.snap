// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelImage component should render HotelImage component correctly 1`] = `
"<Fragment>
  <div className=\\"overlay\\" onClick={[Function: onClick]}>
    <span className=\\"overlay__text\\">
      Click
    </span>
  </div>
  <img className=\\"hotel__image\\" src=\\"h\\" alt=\\"zyx\\" />
</Fragment>"
`;
